cli/EAP7-737_CLI_match open_close_characters.adoc

```diff
@@ -0,0 +1,61 @@
+= EAP7-737 Match open/close characters on CLI commands
+:author:            Ingo Weiss
+:email:             ingo@redhat.com
+:toc:               left
+:icons:             font
+:keywords:          comma,separated,tags
+:idprefix:
+:idseparator:       -
+:issue-base-url:    https://issues.jboss.org/browse/
+
+== Overview
+
+When using the CLI to configure WildFly, sometimes commands with multiple options and parameters require the use of `( )`, `[ ]` and `{ }`. This becomes hard to read once there are line breaks for the same command. Matching the opening and closing characters of these pairs help legibility
+
+In a context where we have a nesting of complex objects as operation parameters, it is difficult to match `(`, `{` and `[` with their closing counterparts. In particular at the end of the command, it is needed to quickly identify missing closing characters.
+
+We are proposing here to highlight the matching pairs when the cursor is on the closing or opening character.
+
+== Issue Metadata
+
+=== Issue:
+
+* {issue-base-url}EAP-737[EAP-737]
+
+=== Related issues:
+
+* {issue-base-url}WFCORE-2663[WFCORE-2663]
+
+=== Dev Contacts:
+
+* mailto:ingo@redhat.com[Ingo Weiss]
+* mailto:jdenise@redhat.com[Jean-François Denise]
+
+=== QE Contacts:
+
+* tbd
+
+=== Affected Projects or Components:
+
+* WildFly Core CLI
+
+== Requirements
+
+=== Hard requirements
+
```

Comment:
As a hard requirement I would add that when exiting the CLI process the current line (can occurs withCtrl-C) as to be cleared from coloured content.

Reply:
Sorry but could you explain more? Right now Ctrl+C just quits the CLI or cancels a paging output.

Reply:
Be sure to clear the colouring in order to not have a remaining coloured character. The branch if I remember well calls matcher.clear().

Reply:
It does, https://github.com/wildfly/wildfly-core/pull/3608/files#diff-7738bbe69f7fc21b71f1d4c7f752f1f6R639